src/test/java/org/alfresco/rest/api/tests/TestFavourites.java

```diff
@@ -1884,6 +1888,42 @@ public void testCreateAndGetFavouriteWithIncludeProperties() throws Exception
         assertNotNull("Properties shouldn't be null because we created some properties while locking the file", file1FavouriteResponse.getProperties());
     }
 
+    /**
+     * REPO-4772 Tests NPE thrown when getting favourites with 'include' properties parameter.
+     *
+     * <p>POST:</p>
+     * {@literal <host>:<port>/alfresco/api/<networkId>/public/alfresco/versions/1/people/<userName>/favorites?include=properties}
+     *
+     * <p>GET:</p>
+     * {@literal <host>:<port>/alfresco/api/<networkId>/public/alfresco/versions/1/people/<userName>/favorites/<targetId>?include=properties}
+     */
+    @Test
+    public void testCreateAndGetFavouriteWithIncludeEmptyPropertiesFile() throws Exception
+    {
+        setRequestContext(networkOne.getId(), networkAdmin, "admin");
+        Map<String, String> include = Collections.singletonMap("include", "properties");
+
+        String f0Id = createFolder(Nodes.PATH_ROOT, "f0-testCreateEmptyFile-"+RUNID).getId();
+        String postUrl = getNodeChildrenUrl(f0Id);
+        Document d1 = new Document();
+        d1.setName("d1.txt");
+        d1.setNodeType(TYPE_CM_CONTENT);
+        HttpResponse response = post(postUrl, toJsonAsStringNonNull(d1), 201);
+        Document documentResp = RestApiUtil.parseRestApiEntry(response.getJsonResponse(), Document.class);
+
+        // Favourite the doc (Test Doc1) using POST
+        Favourite file1Favourite = makeFileFavourite(documentResp.getId());
+        Favourite file1FavouriteResponse = favouritesProxy.createFavourite(networkAdmin, file1Favourite, null);
+
+        // Get the favourited doc using GET and include = properties
+        file1FavouriteResponse = favouritesProxy.getFavourite(networkAdmin, file1FavouriteResponse.getTargetGuid(), include);
+        assertNull("Properties shouldn't be null, they were requested via include=properties", file1FavouriteResponse.getProperties());

```

Comment:
Is this expected to be null? then the message is confusing.

Reply:
You are correct, in this specific test, properties should be null as the created node is empty. I will fix the message.